hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieLockConfig.java

```diff
@@ -35,6 +38,13 @@
 import static org.apache.hudi.common.config.LockConfiguration.DEFAULT_LOCK_ACQUIRE_RETRY_WAIT_TIME_IN_MILLIS;
 import static org.apache.hudi.common.config.LockConfiguration.DEFAULT_ZK_CONNECTION_TIMEOUT_MS;
 import static org.apache.hudi.common.config.LockConfiguration.DEFAULT_ZK_SESSION_TIMEOUT_MS;
+import static org.apache.hudi.common.config.LockConfiguration.DYNAMODB_LOCK_BILLING_MODE_PROP_KEY;
```

Comment:
We have the ability to group the configs on docs together i.e `HoodieLockConfig` + `DynamoDBLockConfig`  so if this is the reasons for the dependency, we can actually move this back to `hoodie-aws`?

Reply:
This part of code is depend on ```HoodieWriteConfig```:
```
public static final ConfigProperty<String> DYNAMODB_LOCK_PARTITION_KEY = ConfigProperty
            .key(DYNAMODB_BASED_LOCK_PROPERTY_PREFIX + "partition_key")
            .noDefaultValue()
            .withInferFunction(cfg -> {
                if (cfg.contains(HoodieWriteConfig.TBL_NAME)) {
                    return Option.of(cfg.getString(HoodieWriteConfig.TBL_NAME));
                }
                return Option.empty();
            })
            .withDocumentation("For DynamoDB based lock provider, the partition key for the DynamoDB lock table. "
                    + "Each Hudi dataset should has it's unique key so concurrent writers could refer to the same partition key."
                    + " By default we use the Hudi table name specified to be the partition key");
```

So I didn't move it to ```hoodie-aws```.

Reply:
if we go with the approach I suggested above, these configs and constants will move to hudi-aws module and we can make hudi-aws depend on hudi-client-common(for the TBL_NAME usage above)